Added MenuItem readme

cburton4 opened the pull request:
## Description
Added documentation to describe the use and functionality of the MenuItem component.

@karmatosed @melchoyce

cburton4 committed:
Added MenuItem readme

youknowriad commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,151 @@
+
+![An image of a MenuItem being highlighted inside of a DropdownMenu component](https://wordpress.org/gutenberg/files/2018/11/MenuItem.png)
+
+1. MenuItem
```
Not sure why do we need this?

cburton4 replied:
Hey! Thanks! I'm excited to get started. Thank you for taking a look and leaving feedback. Greatly appreciated. 

> Not sure why do we need this?

Are you referring to the image that I included? The thinking was that it could be helpful for folks to have a visual included of how the component is used. I'd love to hear your additional thoughts here.

youknowriad replied:
Oh sorry for the confusion, the README is great :). I was actually talking about this particular line `1. MenuItem`. It felt useless?

cburton4 replied:
The thinking was that the `<1. MenuIte>`  label corresponds to the 1 in the image diagram. It's less helpful in this example but in other examples there could be multiple items in an image that we want to clarify and this labeling would help. We could remove it for this example because there is only one item, but it could be useful to establish the pattern for other examples.

What do you think?

youknowriad replied:
Oh I totally missed the diagram while looking at the diff 🙃 

So yes, it's definitely worth keeping. Thanks for the clarification.

youknowriad commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,151 @@
+1. MenuItem
+
+
+
```
I see a lot of empty lines, Shouldn't we just separate sections with a single empty line?

cburton4 replied:
Yep! Sounds great. I'll update that.

youknowriad commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,151 @@
+
+
+
+##Design guidelines
```
This heading (and the followings) are missing a space after the `##`

cburton4 replied:
Whoops! Thank you, will update.

youknowriad commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,151 @@
+
+###Usage
+
+The MenuItem button should only be used in combination with the DropdownMenu component.
```
I'm not certain this is true. I think the `MenuItem` should be used in combination with `MenuGroup` and it can also go inside `Dropdown` which is a less opinionated `DropdownMenu`. Actually, I think it can't be used with `DropdownMenu` at the moment.

cburton4 replied:
Ahhh okay that is helpful. I'll update and to make this more accurate.

talldan replied:
I just wanted to mention that a `MenuItem` should really only be used with a `MenuGroup`. The reason behind that is their accessibility roles work together (`MenuGroup` by default outputs a role of 'menu', and `MenuItem` outputs a role of 'menuitem'. The `Dropdown` component doesn't output a role afaik).

A `MenuGroup` containing `MenuItem`s can be used within a `Dropdown`. A `MenuGroup` can also have other `MenuGroup`s within it so that menus can be nested.

youknowriad reviewed (commented):
Welcome to the Gutenberg repository. This is a great PR with a great start. I left some comments.

cburton4 committed:
Updated spacing

Updated the spacing to remove unneccesary lines.

cburton4 committed:
Updated usage

Updated details for how and when to use the MenuItem component.

chrisvanpatten commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,117 @@
+
+## Design guidelines
+
+###Usage
```
space between # and U

cburton4 replied:
Same here. I believe this is updated.

chrisvanpatten commented on docs/designers-developers/designers/menu-item.md:
````diff
@@ -0,0 +1,117 @@
+```
+import { MenuItem } from '@wordpress/components';
+import { withState } from '@wordpress/compose';
+```
````
I think the import statement code block should be grouped with the below code block instead of being separate

chrisvanpatten commented on docs/designers-developers/designers/menu-item.md:
````diff
@@ -0,0 +1,117 @@
+```children```
+
+- Type: WPElement
+
````
newline not necessary

cburton4 replied:
Same here. I believe this is updated.

chrisvanpatten commented on docs/designers-developers/designers/menu-item.md:
````diff
@@ -0,0 +1,117 @@
+
+`MenuItem` supports the following props. Any additional props are passed through to the underlying  `Button` or `IconButton` component.
+
+```children```
````
this should only be one backtick, not 3

cburton4 replied:
Same here. I believe this is updated.

chrisvanpatten commented on docs/designers-developers/designers/menu-item.md:
````diff
@@ -0,0 +1,117 @@
+
+Element
+
+```label```
````
one backtick, not 3

chrisvanpatten commented on docs/designers-developers/designers/menu-item.md:
````diff
@@ -0,0 +1,117 @@
+
+Defaults to the value of children, if children is passed as a string.
+
+```info```
````
one backtick, not 3

chrisvanpatten reviewed (changes requested):
Partial review but there are some formatting issues with extra newlines, code block formatting instead of inline code formatting, etc. I didn't mark every instance of these problems but tried to mark each _type_ of issue. Can you audit the full file for all cases of these and update?

Thanks :) Excited to see more component documentation!

chrisvanpatten commented:
@melchoyce totally happy with existing documentation to either update all at once, or iteratively :) Just want to make sure new stuff that lands is good, but I’m not too bothered if we don’t have old stuff all fixed at once.

melchoyce commented:
_(hi I thought this was #11872, nevermind, sorry @cburton4)_

cburton4 committed:
Updated formatting errors

I believe I updated formatting errors in spacing, extra spaces as well as backticks.

cburton4 commented:
Hey @chrisvanpatten! Thank you for your review. I believe I updated the errors you found. Hey @jasmussen can you help me to make sure this is formatted correctly?

Thank you both!

cburton4 commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,117 @@
+
+## Development guidelines
+
+###Usage
```
I believe this is updated. Was using Typora before but switched to Sublime so I could see the formatting more clearly.

cburton4 commented on docs/designers-developers/designers/menu-item.md:
```diff
@@ -0,0 +1,117 @@
+2. [Development guidelines](http://#development-guidelines)
+3. [Related components](http://#related-components)
+
+
```
Same here. I believe this is updated.

jasmussen committed:
Fix up a few things!

- Ticks
- Restored an older version as there was a slight mixup
- Whitespace
- Other polish

jasmussen commented:
Okay, thank you for your patience and contributions, Courtney! I pushed a little polish in https://github.com/WordPress/gutenberg/pull/12485/commits/85de8ddf495eb0e6f7fb945ac8880de4246696f3, and I think this should be good to go, perhaps with another set of eyes from @chrisvanpatten.

cburton4 commented:
Thank you @jasmussen!

cburton4 commented:
@chrisvanpatten does this look good to merge?

cburton4 commented:
Sorry! Just realized that Joen already pinged you 😄:Puts head in hands:

ajitbohra commented:
would be good to have example in both `es5` & `es6`

jasmussen commented:
ES5 and ES6 examles would be nice, but I also think this is something we can add as we go, i.e. worth doing separately. It would be nice to get going with this one, I would imagine many other READMEs can follow!

chrisvanpatten commented:
Just seeing my pings - will review today!

chrisvanpatten reviewed (approved):
Looks great to me!

chrisvanpatten commented:
The Travis failure is for an E2E test so I'm going to force-merge this. I promise I looked closely this time 😅

cburton4 commented:
@chrisvanpatten thank you!
